components/renku_data_services/data_connectors/db.py

```diff
@@ -478,11 +478,74 @@ def __init__(
         data_connector_repo: DataConnectorRepository,
         user_repo: UserRepo,
         secret_service_public_key: rsa.RSAPublicKey,
+        authz: Authz,
     ) -> None:
         self.session_maker = session_maker
         self.data_connector_repo = data_connector_repo
         self.user_repo = user_repo
         self.secret_service_public_key = secret_service_public_key
+        self.authz = authz
+
+    async def get_data_connectors_with_secrets(
+        self,
+        user: base_models.APIUser,
+        project_id: ULID,
+    ) -> AsyncIterator[models.DataConnectorWithSecrets]:
+        """Get all data connectors and their secrets for a project."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+
+        can_read_project = await self.authz.has_permission(user, ResourceType.project, project_id, Scope.READ)
+        if not can_read_project:
+            raise errors.MissingResourceError(
+                message=f"The project ID with {project_id} does not exist or you dont have permission to access it"
+            )
+
+        async with self.session_maker() as session:
+            stmt = (
```

Comment:
Some 2 cents, feel free to ignore: it feels like it would be simpler to update the ORM classes to allow for the secrets to be selectively retrieved here, instead of doing this complex query by hand.

Reply:
I assume you mean you just use `relationship` on the data connector ORM class so that you can then query for the project connectors and for the secrets? If yes then I added this.

Reply:
It looks better now, though the second `.where()` seems superfluous.